crates/maze/src/maze.cairo

```diff
@@ -0,0 +1,614 @@
+            0 => {
+                let mut new_position = position + self.width;
+                self.grid = Self::set(self.grid, new_position);
+                if !edge {
+                    new_position = position + 2 * self.width;
+                    self.grid = Self::set(self.grid, new_position);
+                }
+                new_position
+            },
+            1 => {
+                let mut new_position = position + 1;
+                self.grid = Self::set(self.grid, new_position);
+                if !edge {
+                    new_position = position + 2;
+                    self.grid = Self::set(self.grid, new_position);
+                }
+                new_position
+            },
+            2 => {
+                let mut new_position = position - self.width;
+                self.grid = Self::set(self.grid, new_position);
+                if !edge {
+                    new_position = position - 2 * self.width;
+                    self.grid = Self::set(self.grid, new_position);
+                }
+                new_position
+            },
+            _ => {
+                let mut new_position = position - 1;
+                self.grid = Self::set(self.grid, new_position);
+                if !edge {
+                    new_position = position - 2;
+                    self.grid = Self::set(self.grid, new_position);
+                }
+                new_position
+            },
+        }
+    }
+
+    #[inline]
+    fn compute_shuffled_directions(seed: felt252) -> u32 {
+        // [Compute] Random number
+        let mut random: u32 = (seed.into() % 24_u256).try_into().unwrap();
+        // [Return] Pickup a random permutation
+        // [Info] 0:Top, 1:Right, 2:Bottom, 3:Left
+        match random {
+            0 => 0x0123,
+            1 => 0x0132,
+            2 => 0x0213,
+            3 => 0x0231,
+            4 => 0x0312,
+            5 => 0x0321,
+            6 => 0x1023,
+            7 => 0x1032,
+            8 => 0x1203,
+            9 => 0x1230,
+            10 => 0x1302,
+            11 => 0x1320,
+            12 => 0x2013,
+            13 => 0x2031,
+            14 => 0x2103,
+            15 => 0x2130,
+            16 => 0x2301,
+            17 => 0x2310,
+            18 => 0x3012,
+            19 => 0x3021,
+            20 => 0x3102,
+            21 => 0x3120,
+            22 => 0x3201,
+            _ => 0x3210,
+        }
+    }
+
+    #[inline]
+    fn get(value: felt252, index: u8) -> u8 {
+        let value: u256 = value.into();
+        let offset: u256 = Self::two_power(index);
+        (value / offset % 2).try_into().unwrap()
+    }
+
+    #[inline]
+    fn set(value: felt252, index: u8) -> felt252 {
+        // [Info] Unsafe since the value at index is expected to be null
+        let value: u256 = value.into();
+        let offset: u256 = Self::two_power(index);
+        (value + offset).try_into().unwrap()
+    }
+
+    #[inline]
+    fn two_power(exp: u8) -> u256 {
```

Comment:
gm ser, nice maze!

 for powers of two you can now use lookup tables from 2.7.0, which should be easier than this! And you can also use a macro `pow!` from `alexandria_macros` !